custom_models/import_models/llama-3/llama3-sftt-llama3-fine-tuning.ipynb

```diff
@@ -0,0 +1,664 @@
+    "\n",
+    "This notebook is inspired by Philipp Schmid Blog - https://www.philschmid.de/fsdp-qlora-llama3"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5da662ec",
+   "metadata": {},
+   "source": [
+    "## Model License information"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2541e27d",
+   "metadata": {},
+   "source": [
+    "In this notebook we use the Meta Llama3 model from HuggingFace. This model is a gated model within HuggingFace repository. To use this model you have to agree to the license agreement (https://llama.meta.com/llama3/license) and request access to the model before it can be used in this notebook."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3c2860a3-68b3-435d-8d2c-753c90a24da3",
+   "metadata": {},
+   "source": [
+    "### Install the Pre-Requisites"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "885a83fd-92b8-47ad-9128-c820fb4a6188",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install transformers \"sagemaker>=2.190.0\" \"huggingface_hub\" \"datasets[s3]==2.18.0\" --upgrade --quiet\n",
+    "!pip install boto3 s3fs \"aiobotocore==2.11.0\" --upgrade --quiet"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "986ce221-9721-4089-8f5f-9c2d4afba925",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from huggingface_hub import notebook_login\n",
+    "notebook_login()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fe5f5960-12b5-4310-8711-323dde090b96",
+   "metadata": {},
+   "source": [
+    "### Setup\n",
+    "We will initialize the SageMaker Session required to finetune the model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d29cc282-7e49-4db4-ac7f-782b11558395",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sagemaker\n",
+    "import boto3\n",
+    "sess = sagemaker.Session()\n",
+    "# sagemaker session bucket -> used for uploading data, models and logs\n",
+    "# sagemaker will automatically create this bucket if it not exists\n",
+    "sagemaker_session_bucket=None\n",
+    "if sagemaker_session_bucket is None and sess is not None:\n",
+    "    # set to default bucket if a bucket name is not given\n",
+    "    sagemaker_session_bucket = sess.default_bucket()\n",
+    " \n",
+    "try:\n",
+    "    role = sagemaker.get_execution_role()\n",
+    "except ValueError:\n",
+    "    iam = boto3.client('iam')\n",
+    "    role = iam.get_role(RoleName='sagemaker_execution_role')['Role']['Arn']\n",
+    " \n",
+    "sess = sagemaker.Session(default_bucket=sagemaker_session_bucket)\n",
+    " \n",
+    "print(f\"sagemaker role arn: {role}\")\n",
+    "print(f\"sagemaker bucket: {sess.default_bucket()}\")\n",
+    "print(f\"sagemaker session region: {sess.boto_region_name}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ae52c6b5-984a-4ab4-b9b1-e61feb154e49",
```

Comment:
can we add an explanation of the lora and links and why do we need to do this

Reply:
Added explanation for FSDP and QLORA as well as links.